apis/python/src/tiledbsoma/io/outgest.py

```diff
@@ -344,26 +364,37 @@ def to_anndata(
     # * We could use **kwargs -- but that would bork the online help docs.
     # * Our consolation: check if the layer name is the _default_,
     #   and the experiment doesn't have it.
-    anndata_X_future: Future[Matrix] | None = None
+    anndata_X_future: Future[Matrix] | "da.Array" | None = None
```

Comment:
the typing and Future|da.Array  polymorphism was applied to the `X` slot, but not to the handling of `layers`.  But the code below (lines 391) can assign a da.Array to the anndata_layers_futures dict as well, so it needs to be properly typed and resolved.